Installing on Ubuntu(latest) through a github action workflow does not work

I am trying to install the cli in a worker that github action spawns to be able to create issues or pull requests.
I followed the instructions for Ubuntu in the Readme. However it gives me the following error -
**Log**  
```
.
.
.
 Preparing to unpack .../archives/gh_1.2.1_amd64.deb ...
Unpacking gh (1.2.1) over (1.2.0) ...
Setting up gh (1.2.1) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
accepts 0 arg(s), received 1
Error: Process completed with exit code 1.
```

**Relevant workflow file code for the installation**
```yml
      - name: Install GitHub CLI
        run: |
          sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-key C99B11DEB97541F0
          sudo apt-add-repository https://cli.github.com/packages
          sudo apt update
          sudo apt install gh
          gh auth login --with-token ${{ secrets.GH_CLI_AUTH_TOKEN }}
```

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ MSI installers are available for download on the [releases page][].
 
 ### Github Actions
 
-`gh` is pre-installed on the `ubuntu-latest` [Github Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)
+`gh` is pre-installed on [Github Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)
 
 ### Other platforms
 

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ MSI installers are available for download on the [releases page][].
 
 ### Github Actions
 
-`gh` is pre-installed on [Github Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)
+GitHub CLI comes pre-installed in all [Github-Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).
 
 ### Other platforms
 

--- a/README.md
+++ b/README.md
@@ -69,6 +69,10 @@ For more information and distro-specific instructions, see the [Linux installati
 
 MSI installers are available for download on the [releases page][].
 
+### Github Actions
+
+`gh` is pre-installed on the `ubuntu-latest` [Github Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)
+
 ### Other platforms
 
 Download packaged binaries from the [releases page][].

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ MSI installers are available for download on the [releases page][].
 
 ### Github Actions
 
-GitHub CLI comes pre-installed in all [Github-Hosted Runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).
+GitHub CLI comes pre-installed in all [Github-Hosted Runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners).
 
 ### Other platforms
 

